src/Uno.UI/UI/Xaml/Controls/Primitives/Selector.cs

```diff
@@ -548,6 +548,13 @@ private void OnItemsChanged(NotifyCollectionChangedEventArgs e)
 						SelectedIndex = -1;
 					}
 				}
+				//Prevent SelectedIndex been >= Items.Count
+				var sItem = ItemFromIndex(SelectedIndex);
+
+				if (sItem == null || !sItem.Equals(SelectedItem))
```

Comment:
For the second check, I'm very curious how it could happen...  and maybe it should be considered an error?

Reply:
@carldebilly